script/run-comp-match

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+"""
+A script to run a competition match.
+"""
+
+import os
+import sys
+import shutil
+import argparse
+import tempfile
+import contextlib
+import subprocess
+from typing import Iterator
+from pathlib import Path
+from zipfile import ZipFile
+
+REPO_ROOT = Path(__file__).resolve().parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+import controller_utils  # isort:skip
+
+
+def get_zone_path(zone_id: int) -> Path:
+    robot_file: Path = controller_utils.get_zone_robot_file_path(zone_id)
+    return robot_file.parent
+
+
+def construct_match_data(args: argparse.Namespace) -> controller_utils.MatchData:
+    tlas = [
+        tla if tla != '-' else None
+        for tla in args.tla
+    ]
+
+    return controller_utils.MatchData(
+        args.match_num,
+        tlas,
+        args.duration,
+        recording_config=None,  # use the default
+    )
+
+
+@contextlib.contextmanager
+def temporary_arena_root(suffix: str) -> Iterator[None]:
+    arena_root = controller_utils.ARENA_ROOT
+    with tempfile.TemporaryDirectory(suffix=suffix) as tmpdir_name:
+        print(f"Using {tmpdir_name!r} as the arena")  # noqa:T001
+        os.environ['ARENA_ROOT'] = tmpdir_name
+        controller_utils.ARENA_ROOT = Path(tmpdir_name)
+        try:
+            yield
+        finally:
+            os.environ['ARENA_ROOT'] = str(arena_root)
+            controller_utils.ARENA_ROOT = arena_root
+
+
+def prepare_match(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    controller_utils.get_mode_file().write_text('comp\n')
```

Comment:
Given the data is stripped, we can probably remove the newline here

```suggestion
    controller_utils.get_mode_file().write_text('comp')
```

Reply:
This isn't strictly new -- it's part of the code which was moved from `script/prepare-comp-match`. I'm also tempted to leave this as it is since having a trailing newline generally makes text files easier to work with.